### DISCUSSION TITLE
Running one language from another

### DISCUSSION THREAD

**Discussion by Patrick-ring-motive** [codewars/codewars.com#3253] [Polls]
Is it a valid solution to call a different language to finish a kata? Consider this “Rust” example

```rust
use std::process::Command;

fn even_or_odd(number: i32) -> String {
    let script = format!(r#"
sub even_or_odd {{
    my ($number) = @_;
    if ($number % 2) {{
        return "Odd";
    }} else {{
        return "Even";
    }}
}}

print(even_or_odd({}));
1;
"#, number);

    let output = Command::new("perl")
        .arg("-e")
        .arg(script)
        .output()
        .expect("Even");

    let result = String::from_utf8(output.stdout)
        .expect("Odd");

    return result.trim().to_string()
}
```

I could do this to easily rank up to 2kyu in Rust when my actual ability is maybe 5kyu at best.

**Comment by nourdev75536** [comment]
Using a different language or external script to solve a problem in a coding kata is generally considered a form of cheating, particularly if the goal is to demonstrate proficiency in the language specified by the kata. Here’s a breakdown of the options you mentioned:

1. **Yes, this is clever**: While using an external script might be seen as creative, it doesn’t align with the spirit of practicing and improving your skills in the language you're meant to be using. 

2. **No, this is cheating**: This is the most accurate perspective. The intention of coding katas is to test and improve your skills in a specific programming language. Bypassing the challenge by calling code written in another language undermines the learning process.

3. **It’s valid but lame**: Some might see it as valid in a technical sense, but it’s not in line with the purpose of the exercise, which is to practice coding skills in the specified language. It could be considered a "lame" solution because it doesn’t genuinely demonstrate proficiency in the language you’re meant to be using.

4. **I don’t care**: This approach suggests that you’re indifferent to the integrity of the exercise, focusing solely on the end result rather than the learning process.

In summary, the best approach is to solve the problem using the language specified in the kata to genuinely enhance your skills and demonstrate your proficiency. Using external scripts or different languages might help you achieve the result quickly but doesn’t contribute to your growth in the target language.

**Reply by Patrick-ring-motive** [reply]
Do you have a source for what things generally constitute cheating? Besides the overly obvious things like exploiting the test framework and copying others solutions, there doesn't seem to be a good way to tell what some might consider clever solutions and what others might consider cheating. 

I would also argue that there are valuable things to learn by solving a kata this way. It certainly has helped be grow into languages that are new to me because you have to learn how to serialize and deserialize data going both directions and figure out how to interface with an external script. That last part can be quite a task in many languages. Hell I only tried out Haskell just to see if it was possible.

**Comment by hobovsky** [comment]
Whether something is cheating or not is a matter of rules, and not of personal opinion, so I am not even sure why this question is a poll :)

Generally, calling interpreters of other languages from a solution is OK and it is not considered cheating, even if might be seen cheese and lame. It is not cheating to call `python.exe` with an ad-hoc composed script to perform big integer arithmetic from your C code. Exception is when the author of the kata in question predicted such possibility, and blocked possibility of running external executables. In such case, working around restrictions imposed by tests can be considered dishonest and unfair, and can lead to a penalty.
